backend/src/api/database-migration.ts

```diff
@@ -0,0 +1,176 @@
+import config from '../config';
+import { DB } from '../database';
+import logger from '../logger';
+
+class DatabaseMigration {
+  private static currentVersion = 1;
+  private queryTimeout = 120000;
+
+  constructor() { }
+
+  public async $initializeOrMigrateDatabase(): Promise<void> {
```

Comment:
@softsimon I don't like how this method has multiple responsibilities. IMO it should be split into `initializeDatabase` and `migrateDabase` so we know exactly what's being done.